sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -155,6 +155,16 @@ private[spark] object HiveUtils extends Logging {
     .booleanConf
     .createWithDefault(true)
 
+  val HIVE_TABLE_SCAN_MAX_PARALLELISM = buildConf("spark.sql.hive.tableScan.maxParallelism")
```

Comment:
is this really useful? The parallelism should depend on data size, and it's a hard job to tune this config.

Reply:
When reading a Hive partitioned table, users could get an unreasonable number of partitions like dozens of thousands. 

Hive Scan node returns a UnionRDD of Hive partitions. Each Hive partition is read as a HadoopRDD. For each partition, the number depends on data size. But final UnionRDD sums up all number of partitions.

Reply:
is it possible to get the size of each hadoop RDD and do coalesce automatically?

Reply:
I think we can get split size and total number of splits of a Hadoop RDD.